src/sagas/development.saga.test.js

```diff
@@ -0,0 +1,36 @@
+import { call, put, select, takeEvery } from 'redux-saga/effects';
+
+import rootSaga, { showResetDialog } from './development.saga';
+
+import { SHOW_RESET_STATE_PROMPT, resetAllState } from '../actions';
+
+describe('development saga', () => {
+  describe('root development saga', () => {
+    it('should watch for start actions', () => {
+      const saga = rootSaga();
+      expect(saga.next().value).toEqual(
+        takeEvery(SHOW_RESET_STATE_PROMPT, showResetDialog)
+      );
+    });
+  });
+
+  describe('Reset state', () => {
+    it('should reset state to initialState', () => {
+      if (!window.electronStore) {
+        window = { electronStore: { clear: jest.fn() } };
+      }
+
+      const saga = showResetDialog();
+
+      // Show dialog
+      saga.next();
+
+      // Confirm dialog & trigger resetAll
+      expect(saga.next(0).value).toEqual(put(resetAllState()));
+
+      // Check that electronStore.clear is called
+      // Note: No need to check that the config.json is reset as this is tested in the library - just test that it is called
+      expect(window.electronStore.clear).toBeCalled();

```

Comment:
If we do the change I suggest earlier, we'll be calling the method on an instance created in that file, so it won't be so easy to mock.

Happily, that's what `call` is for. I think we can do that by:

```js
const electronStore = new ElectronStore();
expect(saga.next().value).toEqual(
  call([electronStore, electronStore.clear])
);
```

It is technically a different instance but I _think_ this should work. If it doesn't, you can check what `saga.next().value` holds, and just test one of its properties.